### PR TITLE
Incremental SizeOf

### PR DESCRIPTION
**Pull request by imalsogreg**
PR checklist:

* [x] Test coverage for the proposed changes
* [ ] PR description contains example output from repl interaction or a snippet from unit test output
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/kadena-io/pact-core/blob/master/CHANGELOG.md)

Additionally, please justify why you should or should not do the following:

* [ ] Confirm replay/back compat
* [ ] Benchmark regressions
* [ ] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206478668995145

### CODE REVIEW

File: bench/Bench.hs [hunk 1]
```diff
@@ -64,39 +58,43 @@ main = do
   let es = def
   !module1 <- expectEval ee es $ getModule exampleModule1
   coinModule <- expectEval ee es $ getModule coinModuleCode
+
+  -- The comments list the time that each action takes, and the
+  -- number of calls made to the countBytes function.
+  -- In a previous version of the code, these calls were counted in
+  -- the EvalState monad. We do not keep that code in the `master` branch
+  -- because it is messy, but if you need it, it is the last commit of
+  -- the `greg/track-count-bytes` branch.
```
Comment (rsoeldner) [inline bench/Bench.hs:67]: :+1:

File: pact-tng.cabal [hunk 2]
```diff
@@ -129,7 +129,7 @@ library
     cbits/musl/sqrt.c
     cbits/musl/sqrt_data.c
 
-  other-modules: PackageInfo_pact_tng, Paths_pact_tng
+  other-modules: PackageInfo_pact_tng
```
Comment (rsoeldner) [inline pact-tng.cabal:132]: Why this change?

File: pact/Pact/Core/Compile.hs [hunk 3]
```diff
@@ -217,7 +215,7 @@ evalTopLevel bEnv tlFinal deps = do
       let deps' = M.filterWithKey (\k _ -> S.member (_fqModule k) deps) (_loAllLoaded lo0)
           mdata = ModuleData m deps'
       mSize <- sizeOf SizeOfV0 m -- TOOD: Why do we assume SizeOfV0 here?
```
Comment (rsoeldner) [inline pact/Pact/Core/Compile.hs:217]: I dont see a reason this has to be `SizeOfV0`. Maybe @jmcardon has some argument for it?

File: pact/Pact/Core/Compile.hs [hunk 4]
```diff
@@ -233,7 +231,7 @@ evalTopLevel bEnv tlFinal deps = do
       let deps' = M.filterWithKey (\k _ -> S.member (_fqModule k) deps) (_loAllLoaded lo0)
           mdata = InterfaceData iface deps'
       ifaceSize <- sizeOf SizeOfV0 iface -- TOOD: Why do we assume SizeOfV0 here?
```
Comment (rsoeldner) [inline pact/Pact/Core/Compile.hs:233]: same here

File: pact-tng.cabal (hunk 2 above)
Comment (imalsogreg) [reply pact-tng.cabal]: Fixed. No change now.

**Review by rsoeldner — APPROVED** [review]
In general fine, but please address the open questions

**Comment by jmcardon** [comment]
Superceded by #128
